examples/todoapp/lib/base/repositories/todo_repository.dart

```diff
@@ -1,45 +1,204 @@
+import 'dart:async';
+
+import 'package:realm/realm.dart' show Uuid;
+import 'package:rxdart/rxdart.dart';
 import '../common_mappers/error_mappers/error_mapper.dart';
+import '../data_sources/local/connectivity_data_source.dart';
+import '../data_sources/local/todo_local_data_source.dart';
 import '../data_sources/remote/todos_remote_data_source.dart';
 import '../models/todo_model.dart';
+import '../utils/handle_error_mixin.dart';
 
-class TodoRepository {
-  TodoRepository(this._errorMapper, this.dataSource);
+class TodoRepository with ErrorHandlingMixin {
+  TodoRepository(
+    this._errorMapper,
+    this.dataSource,
+    this.localDataSource,
+    this.connectivityDataSource,
+  ) {
+    _connectivitySubscription =
+        connectivityDataSource.connected().listen((event) {
+      if (event) {
+        syncronize();
+      } else {
+        localDataSource.pauseSync();
+      }
+    });
+  }
 
   final ErrorMapper _errorMapper;
   final TodosRemoteDataSource dataSource;
+  final TodoLocalDataSource localDataSource;
+  final ConnectivityDataSource connectivityDataSource;
+  StreamSubscription<bool>? _connectivitySubscription;
 
-  Future<List<TodoModel>> fetchAllTodos() => _errorMapper.execute(
-        () => dataSource.getAllTodos(),
-      );
+  Stream<List<$TodoModel>> fetchAllTodos() => Rx.combineLatest2(
+          _errorMapper.executeStream(localDataSource.allTodos()),
+          _errorMapper
```

Comment:
Cloud you please explain why we need to get the todos from the API?

Reply:
Ah right, the latest todo data will be retrieved from the mongo atlas.
In this case it's not needed